client/material/portal.tsx

```diff
@@ -52,19 +52,35 @@ export function Portal(props: PortalProps) {
     [onDismiss],
   )
 
+  // We don't use the capture/bubble trick for right-clicks since we want to show the context menu
+  // every time they happen, even if it's on the same element.
+  const onContextMenu = useCallback(
+    (event: MouseEvent) => {
+      if (onDismiss) {
+        if (!portalRef.current?.contains(event.target as Node)) {
+          markEventAsHandledDismissal(event)
+          onDismiss()
+        }
+      }
+    },
+    [onDismiss, portalRef],
+  )
+
   useEffect(() => {
     if (open) {
       document.addEventListener('click', onCaptureClick, true /* useCapture */)
       document.addEventListener('click', onBubbleClick)
+      document.addEventListener('contextmenu', onContextMenu, true /* useCapture */)
```

Comment:
Unsure why you don't think we need a bubbling event handler here? This stuff is meant to handle cases where e.g. we have an overlay in a dialog, and they click outside of the dialog to dismiss it. Without that handling, we're gonna dismiss the dialog as well, so please handle it in the same way like we talked about

Reply:
My belief was that the `markEventAsHandledDismissal(event)` line was in charge of that logic, i.e. not dismissing the dialog if there's another overlay in it. And as for the capture/bubble handlers combo, I thought that was done so when you click on user's entry in channel list, it opens their overlay, and then when you click on their entry again, it should close it, which it does. And this is specifically what we don't want to happen when right-clicking, or at least I thought so.

For what it's worth, I did test pretty much all scenarios I could think of and I thought this is behaviour we'd want (e.g. it doesn't dismiss the dialog when they click outside it with another overlay open). The only thing that I noticed that is perhaps a bit weird is when you right-click on same user entry twice, there is no open/close animation the second time since it closes and opens the same overlay too fast I guess? But it seemed like a minor thing to me.